MekHQ/src/mekhq/AtBGameThread.java

```diff
@@ -240,6 +241,16 @@ public void run() {
                         entity.setForceString(force.getFullMMName());
                     }
                     entities.add(entity);
+                    
+                    // if we've swapped this entity in for a bot-controlled unit, copy the bot controlled unit's
+                    // deployment parameters to this entity.
+                    if (scenario instanceof AtBDynamicScenario &&
+                            ((AtBDynamicScenario) scenario).getPlayerUnitSwaps().
+                                containsKey(UUID.fromString(entity.getExternalIdAsString()))) {
```

Comment:
We should probably null check the external id, as it is nullable IIRC

Reply:
I'm pretty sure every player and bot unit has an external ID. The bot units get one upon generation, while the player units always set the corresponding entity's external ID to the unit's ID.